sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java

```diff
@@ -96,6 +108,117 @@ CREATE TABLE series (
         this.addRustTestCase("latenessTest", compiler, getCircuit(compiler), data);
     }
 
+    Long[] profile(String program) throws IOException, InterruptedException {
+        // Rust program which profiles the circuit.
+        String main = """
+               use dbsp::{
+                   algebra::F64,
+                   utils::Tup2,
+                   zset,
+               };
+
+               use sqllib::{
+                   append_to_collection_handle,
+                   read_output_handle,
+                   casts::cast_to_Timestamp_s,
+               };
+
+               use std::{
+                   io::Write,
+                   ops::Add,
+                   fs::File,
+                   time::SystemTime,
+               };
+
+               use temp::circuit;
+
+               #[test]
+               // Run the circuit generated by 'circuit' for a while then measure the
+               // memory consumption.  Write the time taken and the memory used into
+               // a file called "mem.txt".
+               pub fn test() {
+                   let (mut circuit, streams) = circuit(2).expect("could not build circuit");
+                   let start = SystemTime::now();
+
+                   // Initial data value for timestamp
+                   let mut timestamp = cast_to_Timestamp_s("2024-01-10 10:10:10".to_string());
+                   for i in 0..1000000 {
+                       let value = Some(F64::new(i.into()));
+                       timestamp = timestamp.add(20000);
+                       let input = zset!(Tup2::new(value, timestamp) => 1);
+                       append_to_collection_handle(&input, &streams.0);
+                       if i % 100 == 0 {
+                           let _ = circuit.step().expect("could not run circuit");
+                           let _ = &read_output_handle(&streams.1);
+                       }
+                   }
+                   let profile = circuit.retrieve_profile().expect("could not get profile");
+                   let end = SystemTime::now();
+                   let duration = end.duration_since(start).expect("could not get time");
+
+                   let mut data = String::new();
+                   data.push_str(&format!("{},{}\\n",
+                                          duration.as_millis(),
+                                          profile.total_used_bytes().unwrap().bytes));
+                   let mut file = File::create("mem.txt").expect("Could not create file");
+                   file.write_all(data.as_bytes()).expect("Could not write data");
```

Comment:
You could use [`std::fs::write`](https://doc.rust-lang.org/stable/std/fs/fn.write.html) here.

Reply:
Well, it's mostly Rust, so I thought to bother you. I rotate Rust reviewers.
Thanks.

Reply:
But I expect that at some point we will built a monitoring agent for pipelines based on a similar skeleton, exposing virtual views with program statistics.